ILRepack/Steps/SourceServerData/SourceServerDataRepackStep.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+using SourceLink;
+
+namespace ILRepacking.Steps.SourceServerData
+{
+    internal class SourceServerDataRepackStep : IRepackStep
+    {
+        private readonly string _targetPdbFile;
+
+        private readonly IEnumerable<string> _assemblyFiles;
+
+        private byte[] _srcSrv;
+
+        public SourceServerDataRepackStep(string targetFile, IEnumerable<string> assemblyFiles)
+        {
+            if (targetFile == null) throw new ArgumentNullException(nameof(targetFile));
+            if (assemblyFiles == null) throw new ArgumentNullException(nameof(assemblyFiles));
+            _targetPdbFile = PdbPath(targetFile);
+            _assemblyFiles = assemblyFiles;
+        }
+
+        public void Perform()
+        {
+            if (_assemblyFiles.Any())
+            {
+                var pdbs = _assemblyFiles.Select(PdbPath).ToArray();
+                string primaryPdb = pdbs.First();
+                if (File.Exists(primaryPdb))
+                {
+                    _srcSrv = PdbFile.readSrcSrvBytes(primaryPdb);
+                    HttpSourceServerDescriptor primaryDescriptor;
+                    if(HttpSourceServerDescriptor.TryParse(_srcSrv, out primaryDescriptor))
```

Comment:
even if the primary has no pdb/srcsrv, we could still generate one from others AFAICT